modules/stochastic_tools/src/actions/StochasticToolsAction.C

```diff
@@ -65,24 +66,62 @@ StochasticToolsAction::act()
   }
 
   // [Problem]
-  else if (_current_task == "auto_create_problem" && getParam<bool>("auto_create_problem") &&
-           !_awh.hasActions("create_problem"))
+  else if (_current_task == "auto_create_problem" && getParam<bool>("auto_create_problem"))
   {
-    // Build the Action parameters
-    InputParameters action_params = _action_factory.getValidParams("CreateProblemAction");
+    if (_awh.hasActions("create_problem"))
+    {
+      const auto & actions = _awh.getActionListByName("create_problem");
+      for (const auto & act : actions)
+      {
+        CreateProblemAction * p = dynamic_cast<CreateProblemAction *>(act);
+        if (!p)
+          continue;
+        InputParameters & params = p->getObjectParams();
 
-    // Create the action
-    auto action = std::static_pointer_cast<MooseObjectAction>(
-        _action_factory.create("CreateProblemAction", "Problem", action_params));
+        if (params.isParamSetByUser("solve"))
+        {
+          if (params.get<bool>("solve"))
+            mooseError("Problem/solve parameter cannot be set to true for STM main input file.");
+        }
+        else
+          params.set<bool>("solve") = false;
```

Comment:
Don't throw an error. The user most likely know exactly what they are doing if this is set.
```suggestion
        if (!params.isParamSetByUser("solve"))
          params.set<bool>("solve") = false;
```